include/rfc1738.h

```diff
@@ -64,6 +64,9 @@ extern char *rfc1738_do_escape(const char *url, int flags);
  */
 extern void rfc1738_unescape(char *url);
 
+/// convert a single-digit hex ascii character to its corresponding value
+extern int HextDigitToInt(char);
```

Comment:
IMO the reduction of future dependencies on this deprecated code is worth duplicating the logic in this helper function in the `anyp/Uri.cc` where the new decoder is. Leaving the `include/` and `lib/` code untouched.